src/gg.regression.unity.bots/Runtime/Scripts/RGSettings.cs

```diff
@@ -70,35 +62,13 @@ public static RGSettings GetOrCreateSettings()
                 _settings.rgHostAddress = "https://play.regression.gg";
                 _settings.logLevel = DebugLogLevel.Info;
 
-#pragma warning disable CS0618 // Type or member is obsolete
-                _settings.numBots = 0;
-                _settings.botsSelected = Array.Empty<long>();
-#pragma warning restore CS0618 // Type or member is obsolete
-
-                _settings.nextBotId = 0;
-                _settings.nextBotInstanceId = 0;
-
                 _settings.feature_StateRecordingAndReplay = false;
 #if UNITY_EDITOR
                 AssetDatabase.CreateAsset(_settings, SETTINGS_PATH);
                 AssetDatabase.SaveAssets();
 #endif
-            }
-
-            // These fields are now obsolete
-#pragma warning disable CS0618 // Type or member is obsolete
-            _settings.numBots = 0;
-            _settings.botsSelected = Array.Empty<long>();
-            _settings.nextBotId = 0;
-            _settings.nextBotInstanceId = 0;
-#pragma warning restore CS0618 // Type or member is obsolete
-
-            // backwards compat for migrating RG devs before we had a single host address field
-            if (string.IsNullOrEmpty(_settings.rgHostAddress))
-            {
-                _settings.rgHostAddress = "https://play.regression.gg";
-#if UNITY_EDITOR
-                AssetDatabase.SaveAssets();
+#if !UNITY_EDITOR
+                Debug.LogWarning("RG settings could not be loaded. Make sure to log into Regression Games within the Unity Editor before building your project. For now, an empty user settings object will be used.");
```

Comment:
Let's use our internal `RGDebug.LogWarning`

Reply:
Funnily enough I actually can't use that here I think, because it would cause an infinite recursion. RGDebug makes a call to get the RGSettings, and if it doesn't exist, it calls this debug function... which in turn calls RGSettings creation. So I think I'll have to keep it like this, since this is the case where the RGSettings doesn't load.

Reply:
Ah - makes sense!